map.go

```diff
@@ -24,7 +24,7 @@ var (
 	ErrKeyNotExist      = errors.New("key does not exist")
 	ErrKeyExist         = errors.New("key already exists")
 	ErrIterationAborted = errors.New("iteration aborted")
-	ErrMapIncompatible  = errors.New("map's spec is incompatible with pinned map")
+	ErrMapIncompatible  = errors.New("map's spec is incompatible")
```

Comment:
I see the initial "map's spec" is/was a bit ambiguous, since a MapSpec is always caller-provided, and doesn't necessarily relate to an existing Map. We could improve this as follows:

```suggestion
	ErrMapIncompatible  = errors.New("map spec is incompatible with existing map")
```

This is clear both in the pinned Map case and in the manually-created Map case (which wasn't necessarily pinned). Good catch. :+1: